activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/validation/JsonSchemaModelValidator.java

```diff
@@ -41,111 +45,145 @@
  */
 public abstract class JsonSchemaModelValidator implements ModelValidator {
 
-    private final Logger log = LoggerFactory.getLogger(JsonSchemaModelValidator.class);
-
-    protected abstract SchemaLoader schemaLoader();
-
-    @Override
-    public void validate(byte[] bytes,
-                         ValidationContext validationContext) {
-        JSONObject processExtensionJson = null;
-        try {
-            log.debug("Validating json model content: " + new String(bytes));
-            processExtensionJson = new JSONObject(new JSONTokener(new String(bytes)));
-            schemaLoader()
-                    .load()
-                    .build()
-                    .validate(processExtensionJson);
-        } catch (JSONException jsonException) {
-            log.debug("Syntactic model JSON validation errors encountered",
-                      jsonException);
-            throw new SyntacticModelValidationException(jsonException);
-        } catch (ValidationException validationException) {
-            log.debug("Semantic model validation errors encountered: " + validationException.toJSON(),
-                      validationException);
-            throw new SemanticModelValidationException(validationException.getMessage(),
-                                                       getValidationErrors(validationException, processExtensionJson));
-        }
+  private final Logger log = LoggerFactory.getLogger(JsonSchemaModelValidator.class);
+
+  protected abstract SchemaLoader schemaLoader();
+
+  @Override
+  public void validate(byte[] modelContent,
+                       ValidationContext validationContext) {
+    internalValidation(modelContent, true);
+  }
+
+  @Override
+  public Collection<ModelValidationError> validateAndReturnErrors(byte[] modelContent,
+                                                                  ValidationContext validationContext) {
+    return internalValidation(modelContent, false);
+  }
+
+  private Collection<ModelValidationError> internalValidation(byte[] modelContent, boolean shouldThrowError) {
+    JSONObject processExtensionJson = null;
+    String jsonContent = new String(modelContent);
+    try {
+      log.debug("Validating json model content: " + jsonContent);
+      processExtensionJson = extractJsonObject(modelContent);
+      validateJson(processExtensionJson);
+    } catch (JSONException jsonException) {
+      log.debug("Syntactic model JSON validation errors encountered",
+                jsonException);
+      if (shouldThrowError) {
+        throw new SyntacticModelValidationException(jsonException);
+      } else {
+        String problem = "Could not parse:" + jsonContent;
+        return List.of(new SyntacticModelError(problem, jsonException.getMessage()));
+      }
```

Comment:
Maybe in this specific case, where we are not able even to parse the json file, it's fine throwing an exception directly, without connecting it in a list, as something really bad happened: the provide file is not a json.

Reply:
Ok, will throw this error